tests/unit/test_business_logic.py

```diff
@@ -438,6 +455,16 @@ def test_set_status_cannot_action_own_request(bll):
         bll.set_status(release_request2, RequestStatus.RELEASED, user=user)
 
 
+def test_set_status_approved_no_files_denied(bll):
+    user = factories.create_user("checker", [], True)
+    release_request1 = factories.create_release_request(
+        "workspace", status=RequestStatus.SUBMITTED
+    )
+
+    with pytest.raises(bll.RequestPermissionDenied):
+        bll.set_status(release_request1, RequestStatus.APPROVED, user=user)
```

Comment:
Maybe also test that a request with only a supporting file can't be released?

Reply:
done